src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -70,13 +70,12 @@ default void copyFileInWorkspace(String sourceFile, String targetFile, FilePath
         targetFilePath.chmod(0755);
     }
 
-    default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName,FilePath workspace)
+    default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName, FilePath workspace)
             throws IOException, InterruptedException {
         /*Use of Computer is not recommended as jenkins hygeine for pipeline support
          * https://javadoc.jenkins-ci.org/jenkins/tasks/SimpleBuildStep.html */
         
-        Computer cmp = workspace.toComputer();
-        String tmpDir = (String) cmp.getSystemProperties().get("java.io.tmpdir");
+        String tmpDir = getNodeSpecificTmpFolderPath(workspace);
         if (launcher.isUnix()) {
             tmpDir = tmpDir + "/" + uniqueName;

```

Comment:
I  think we should avoid having our code do this platform specific path separator handling to the extend possible.
hudson.FilePath and java.io.FilePath classes offer the facility of adding to paths and will abstract out the platform differences.

Reply:
Done.